### DISCUSSION TITLE
Troubleshooting bugs

### DISCUSSION THREAD

**Discussion by jejukin** [DarkflameUniverse/DarkflameServer#1144] [Q&A] [answered]
Hi, I've been using darkflame universe at home to host a world for just me and my kids for a while now (a year or so?) and we've played off and on and we love it.

Thank you for sharing this project!  My original build from around a year ago worked pretty well - there was some frustrating glitching with mob pathing, but it was limited to Gnarled Forest.  Around Brig Rock and Maelstrom Trench, the mobs would aggro you on platforms where you should be safe (eg, at the quest-giver in Maelstrom Trench; or at the skull-platform mini-build or any of the platforms on Brig Rock).  The mobs would fly or warp to reach you and attack you no matter where you were.  Super frustrating, but it was only one zone.

That was from a build around a year ago, made on 8/30/2022.  Recently I thought I would update the server and see if it helps; and it did help with that issue.  I could jump all around the platforms at Brig rock and not be attacked, and I could stand at the quest giver on Maelstrom Trench and not be attacked, both of which were nice.  However, the client keeps de-syncing with the server whenever mobs are pathing.  It happens whenever the Admirals are aggro'd in Brig Rock, it happens in Avent Gardens outside the Sentinel base camp.  Basically mobs will just appear to stop aggroing or chasing, but what's really happening is that the client isn't interpreting the events accurately so you're being attacked and you just don't know it until eventually the server catches up and you're smashed.  

In some cases you're not being attacked, like with the pirates at Brig Rock.  They'll aggro you from below and not warp up or fly to kill you, but while they're aggro and their pathing is borked, the client de-syncs and stops responding.  I'll hit shift to interact with the prisoner ninjas and just nothing happens.  If I log out or get far enough away to leash the mob, eventually the de-sync ends and the client and server catch up with eachother, but in the meantime I can smash stuff and no loot drops (until much later, when client and server sync again).  

While the issue from last August was frustrating in Gnarled forest, the new issue is frustrating in lots of places.  Pathing is all screwed up and it de-syncs constantly.  I created a brand-new VM of ubuntu 22-04 with fresh updates and did a clean server install, and it still had this problem.  

On the other hand, I can revert to a snapshot of my old VM and use the build I made in August of last year, and I don't de-sync, I just get murdered a lot on Brig Rock.  But that's at least better than the whole world being messed up.

Has anyone else run into these issues and troubleshot them?

**Comment by EmosewaMC** [comment, marked as answer]
go to build/sharedconfig.ini and change maximum_outgoing_bandwidth from 80000 to 0

**Reply by jejukin** [reply]
Oh wow, that's amazing, thanks for the reply!  I did notice that it was even documented in the sharedconfig.ini, so thanks doubly since I ought to have seen that.

**Reply by EmosewaMC** [reply]
This wouldn't be an issue should we have proper delta compression but alas we do not do much delta compression right now so many bits are wasted.  Hopefully in the future we can operate on a limit.  Time shall tell
